dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/BeanDeploymentArchiveImpl.java

```diff
@@ -590,26 +596,21 @@ public boolean hasBeans() {
     @Trivial
     // This is marked trivial because it's called when iterating over all BDAs to look
     // for a single class which causes lots of noise in the trace without adding any value
-    public Map<String, Class<?>> getAllClazzes() {
+    public Set<String> getAllClazzes() {
         return allClasses;
     }
 
     @Override
     public Collection<String> getBeanClasses() {
-        return beanClasses.keySet();
+        return beanClasses;
     }
 
     @Override
-    public boolean containsBeanClass(Class<?> clazz) {
+    public boolean containsBeanClass(String clazz) {
```

Comment:
I think a ClassLoader param should be added ... it is the unique combination of class name and classloader that you are looking for